JS_README.md

```diff
@@ -0,0 +1,115 @@
+This fork of emacs integrates the [Deno Runtime](https://deno.land/). We do this by integrating Chrome's Javascript engine V8, and the deno_core and deno_runtime libraries. deno is dependent on [tokio](https://github.com/tokio-rs/tokio), so we integrate that as well. This code is a strictly additive layer, it changes no elisp functionality, and should be able to merge upstream patches cleanly.
+
+The high level concept here is to allow full control of emacs-ng via the javascript layer. In order to avoid conflict between the elisp and javascript layers, only one scripting engine will be running at a time. elisp is the "authoritative" layer, in that the javascript will invoke elisp functions to interact with the editor. This is done via a special object in javascript called `lisp`. An example of it's usage would be:
+
+``` js
+let buffer = lisp.get_buffer_create("mybuf");
+```
+
+The lisp object uses reflection to invoke the equivalent of `(get-buffer-create "mybuf")`. In this example, get-buffer-create returns a buffer object. In order to represent this, we use a technique called proxying, in which we give a reference to the elisp object to an internal field of a javascript object. This field is only accessible by the native layer, so it cannot be tampered with in the javascript layer. When javascript calls subsequent functions using `buffer`, a translation layer will extract the elisp object and invoke it on the function.
+
+Primative data structures will be converted automatically. More complex data structures can be converted to native javascript objects via a call to a special function called `json`
```

Comment:
I was thinking about that - seems like for the performance purpose we might need 2 versions of the functions - one that automatically converts and one that uses the original values to avoid redundant conversions. It is not something to address know, but if will be good to know that we can achieve that in the future.

Reply:
We can definitely do that, but I agree I think its something to slate for a future release.